presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/BooleanBlockStateSerializer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.state;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.Type;
+
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+
+class BooleanBlockStateSerializer
+        extends BlockPositionValueStateSerializer<BooleanBlockState>
+{
+    BooleanBlockStateSerializer(Type firstType, Type secondType)
+    {
+        super(firstType, secondType);
+    }
+
+    @Override
+    void readFirstField(Block block, int position, BooleanBlockState state)
+    {
+        state.setFirst(BOOLEAN.getBoolean(block, position));
```

Comment:
Per discussion, use `firstType.getBoolean`; ditto for `writeFirstField` and other serializers.